packages/core/src/components/overlay/overlay.md

```diff
@@ -1,23 +1,36 @@
 @# Overlay
 
-__Overlay__ is a generic low-level component for rendering content _on top of_ its siblings or above the entire
+<div class="@ns-callout @ns-intent-danger @ns-icon-error @ns-callout-has-body-content">
+    <h5 class="@ns-heading">
+
+Deprecated: use [**Overlay2**](#core/components/overlay2)
+
+</h5>
+
+This component is **deprecated since @blueprintjs/core v5.9.0** in favor of the new
+**Overlay2** component which is compatible with React 18 strict mode. You should migrate to the
+new API which will become the standard in Blueprint v6.
```

Comment:
as with other places in the docs, we should avoid stating explicit version numbers in these deprecation notices and say instead "a future major version"